enaml/core/dynamicscope.py

```diff
@@ -0,0 +1,58 @@
+# ------------------------------------------------------------------------------
+# Copyright (c) 2025, Nucleic Development Team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# ------------------------------------------------------------------------------
+from itertools import chain, repeat
+from enaml.core._dynamicscope import _DynamicScope, UserKeyError
+
+
+class DynamicScope(_DynamicScope):
+    """_DynamicScope is a C++ class which exposes the following attributes:
+
+    _owner
+    _change
+    _f_writes
+    _f_locals
+    _f_globals
+    _f_builtins
+
+    """
+
+    def __iter__(self):
+        fwrites_it = iter(self._f_writes or ())
+        self_it = repeat("self", 1)
+        change_it = repeat("change", 1 if self._change else 0)
+        flocals_it = iter(self._f_locals)
+        fglobals_it = iter(self._f_globals)
+        fbuiltins_it = iter(self._f_builtins)
+        fields = set()
+        parent = self._owner
+        while parent is not None:
```

Comment:
Why traverse the parent tree here ahead-of-time instead of allowing `chain` to naturally recurse into the parent iterator as needed?

Reply:
Not sure I follow... the owner is a Declarative, the parent is another Declarative neither are iterable. It could use [traverse_ancestors](https://github.com/nucleic/enaml/blob/main/enaml/core/object.py#L330)?

Since the traceback module uses `list(frame.f_locals)`, if it doesn't remove duplicates it will likely quickly overflow the 750 item limit and not provide a suggestion at all...

Reply:
Or is there a way to get the parent dynamic scope? I don't see it anywhere from a quick look.

Reply:
Sorry, yeah, I had it in my head that `parent` was another `DynamicScope` object. That said, I think we still put all of that logic into a generator function, instead of computing ahead-of-time.

Reply:
For the 750 item limit, that really should be an issued raised with the core traceback module. Instead of 100% punting at 751 names, they should truncate their search space to the first 750 items, and just ignore the rest.